samples/cpp/speech_sample/utils.hpp

```diff
@@ -0,0 +1,395 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include <cnpy.h>
+
+#include <samples/common.hpp>
+
+#define MAX_SCORE_DIFFERENCE 0.0001f  // max score difference for frame error threshold
+#define MAX_VAL_2B_FEAT      16384    // max to find scale factor
+
+typedef std::chrono::high_resolution_clock Time;
+typedef std::chrono::duration<double, std::ratio<1, 1000>> ms;
+typedef std::chrono::duration<float> fsec;
+
+/**
+ * @brief struct to store score error
+ */
+typedef struct {
+    uint32_t numScores;
+    uint32_t numErrors;
+    float threshold;
+    float maxError;
+    float rmsError;
+    float sumError;
+    float sumRmsError;
+    float sumSquaredError;
+    float maxRelError;
+    float sumRelError;
+    float sumSquaredRelError;
+} score_error_t;
+
+/**
+ * @brief struct to store infer request data per frame
+ */
+struct InferRequestStruct {
+    ov::runtime::InferRequest inferRequest;
+    int frameIndex;
+    uint32_t numFramesThisBatch;
+};
+
+/**
+ * @brief Check number of input files and model network inputs
+ * @param numInputs number model inputs
+ * @param numInputFiles number of input files
+ * @return none.
+ */
+void CheckNumberOfInputs(size_t numInputs, size_t numInputFiles) {
+    if (numInputs != numInputFiles) {
+        throw std::logic_error("Number of network inputs (" + std::to_string(numInputs) +
+                               ")"
+                               " is not equal to number of input files (" +
+                               std::to_string(numInputFiles) + ")");
+    }
+}
+/**
+ * @brief Get scale factor for quantization
+ * @param ptrFloatMemory pointer to float memory with speech feature vector
+ * @param targetMax max scale factor
+ * @param numElements number of elements in speech feature vector
+ * @return scale factor
+ */
+float ScaleFactorForQuantization(void* ptrFloatMemory, float targetMax, uint32_t numElements) {
+    float* ptrFloatFeat = reinterpret_cast<float*>(ptrFloatMemory);
+    float max = 0.0;
+    float scaleFactor;
+
+    for (uint32_t i = 0; i < numElements; i++) {
+        if (fabs(ptrFloatFeat[i]) > max) {
+            max = fabs(ptrFloatFeat[i]);
+        }
+    }
+
+    if (max == 0) {
+        scaleFactor = 1.0;
+    } else {
+        scaleFactor = targetMax / max;
+    }
+
+    return (scaleFactor);
+}
+/**
+ * @brief Clean score error
+ * @param error pointer to score error struct
+ * @return none.
+ */
+void ClearScoreError(score_error_t* error) {
+    error->numScores = 0;
+    error->numErrors = 0;
+    error->maxError = 0.0;
+    error->rmsError = 0.0;
+    error->sumError = 0.0;
+    error->sumRmsError = 0.0;
+    error->sumSquaredError = 0.0;
+    error->maxRelError = 0.0;
+    error->sumRelError = 0.0;
+    error->sumSquaredRelError = 0.0;
+}
```

Comment:
Need to add line between end of method and description of next method. This comment is valid for almost methods in this file.

Reply:
Added

Reply:
Was done

Reply:
Was done